sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedOffset.scala

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.spark
+
+import com.azure.cosmos.spark.ChangeFeedOffset.{IdPropertyName, StatePropertyName, V1Identifier}
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.apache.spark.sql.connector.read.streaming.{Offset, PartitionOffset}
+
+private case class ChangeFeedOffset
+(
+  changeFeedState: String
+) extends Offset
+  with Serializable
+  with PartitionOffset
+  with CosmosLoggingTrait {
+
+  logTrace(s"Instantiated ${this.getClass.getSimpleName}")
+
+  @transient private lazy val jsonPersisted = String.format(
+    "{\"%s\":\"%s\",\"%s\":\"%s\"}",
```

Comment:
I wasn't able to get Scala string interpolation to work with the quotes - I can try it again - probably just a question of the right escaping - but the usual variants didn't work - so I used String.format as a fallback where quotes are included.

Reply:
I am guessing that's probably because you tried to break the string interpolation into multiple lines. if you have in one line it should work.


for multiple lines you can do something like see:
```scala
    spark.sql(
      s"""create table myschema (id String, _ts long) using cosmos.oltp options(
        | spark.cosmos.accountEndpoint '${configuration.get("spark.cosmos.accountEndpoint").get}',
        | spark.cosmos.accountKey '${configuration.get("spark.cosmos.accountKey").get}',
        | spark.cosmos.database '${configuration.get("spark.cosmos.database").get}',
        | spark.cosmos.container '${configuration.get("spark.cosmos.container").get}')
      """.stripMargin)
```

Reply:
No - it is really a scala issue
https://github.com/scala/bug/issues/6476

Reply:
No fix has been made yet for scala 2.12 or 2.13

Reply:
But reading through the descripiton I'll see whether I can make it work with raw"whatever" instead of s"whatever"

Reply:
Changed it to string interpolation in product code - but left it as string.format in tests because it is easier to read there. (multiple lines, whitespaces must not show up for comparison etc.)